### PR TITLE
feat: Go feature flag provider

### PR DESCRIPTION
**Pull request by thomaspoignant**
This is a 1st version of the `go-feature-flag` provider.

### CODE REVIEW

**Comment by thomaspoignant** [comment]
@beeme1mr I struggle with the dependencies and where to store them in the mono repo.
I have put them in the `package.json` close to the provider but it is not working for the tests _(see the CI workflow falling)_.

Should I place them on the top level instead? 
Is this something we have to think about for this repo?

**Comment by beeme1mr** [comment]
I'll take a look at your branch and see if I can get the test passing.

**Comment by beeme1mr** [comment]
Hey @thomaspoignant, I was able to get it working. I tried to make my changes in granular commits so you could more easily follow my changes. There's a bit of NX magic involved but it's not too bad. Let me know if you have any questions.

**Comment by thomaspoignant** [comment]
Thanks a lot @beeme1mr for your help 🙏 
I was not used to NX that's the main reason why I implemented bad patterns (such as the dependencies at the wrong level).
I think the PR is now **Ready for review**.

I saw your commits and I wonder if some things should not be part of the `npm run generate-provider` command?
I am referring to https://github.com/open-feature/node-sdk-contrib/pull/64/commits/b54196f0af660bd7440fa11e9c2a33bf4f1f4f7b and https://github.com/open-feature/node-sdk-contrib/pull/64/commits/15b2e8a0a5a93ed620e3946d114caf32ee8a45ff.
This looks like something that will be required for all providers, so I wonder if we should not add this to the templates?

**Comment by beeme1mr** [comment]
https://github.com/open-feature/node-sdk-contrib/pull/64/commits/18c11da223b7a59db318ad5fe3177ba331a73a9b was added to the generator after you ran it. Future providers and hooks will have this setup automatically.

**Comment by toddbaert** [comment]
@thomaspoignant I updated the generator with https://github.com/open-feature/node-sdk-contrib/pull/63/files

It adds some additional scripts to the package.json to enable releasing. I will add this to this PR if you dont mind to make sure all the release process works as expected.

**Comment by toddbaert** [comment]
@thomaspoignant this is done. You can view my changes [on this commit](https://github.com/open-feature/node-sdk-contrib/pull/64/commits/accd42f62601ba3c02cd1c05eb0b64f20762335b)

This stuff is now included in the generator, my commit just backports it to your changes.

File: libs/providers/go-feature-flag/project.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
     "publish": {
       "executor": "@nrwl/workspace:run-commands",
       "options": {
-        "command": "node tools/scripts/publish.mjs providers-go-feature-flag {args.ver} {args.tag}"
+        "command": "npm run publish-if-not-exists",
+        "cwd": "dist/libs/providers/go-feature-flag"
```
Comment (toddbaert) [inline libs/providers/go-feature-flag/project.json:10]: This is the only change in this file, rest is just whitespace.

**Comment by thomaspoignant** [comment]
@toddbaert thanks a lot for your changes.
It seems that the deployment failed https://github.com/open-feature/node-sdk-contrib/runs/7456249885?check_suite_focus=true

I can have a look tomorrow when I will be in front of my computer

**Pull request merged**